Sync feature flags tables and add upgrade page link (#741)
crossplane/docs: content/master/software/install.md

| Status | Flag | Description |
| --- | --- | --- |
| Beta | `--enable-composition-functions` | Enable support for Composition Functions. |

> | Beta | `--enable-composition-functions-extra-resources` | Enable support for Composition Functions Extra Resources. Only respected if --enable-composition-functions is set to true. |

Reviewer: ```suggestion
| Beta | `--enable-composition-functions-extra-resources` | Enable support for Composition Functions Extra Resources. Only respected with `--enable-composition-functions` enabled. |
```
vale fixes

| Beta | `--enable-composition-webhook-schema-validation` | Enable Composition validation using schemas. |

> | Beta | `--enable-deployment-runtime-configs` | Enable support for Deployment Runtime Configs. |

Reviewer: ```suggestion
| Beta | `--enable-deployment-runtime-configs` | Enable support for DeploymentRuntimeConfigs. |
```
vale fix.
All caps for the kind or lowercase if they are separate words.
